bofire/data_models/features/categorical.py

```diff
@@ -363,39 +365,52 @@ class CategoricalOutput(Output):
     order_id: ClassVar[int] = 8
 
     categories: TCategoryVals
-    objective: Annotated[
-        List[Annotated[float, Field(type=float, ge=0, le=1)]], Field(min_items=2)
-    ]
+    objective: Optional[
+        Union[CategoricalObjective, ConstrainedCategoricalObjective]
```

Comment:
Hmm, `CategoricalObjective` is an abstract Objective. We only allow here for the actual ones. You could introduce in `data_models/objectives/api.py` a type called `AnyCategoricalObjective`, which is for now just equal to ConstrainedCategoricalObjective.